ai/anthropic/v0/component_test.go

```diff
@@ -0,0 +1,155 @@
+package anthropic
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/instill-ai/component/base"
+	"github.com/instill-ai/component/internal/util/httpclient"
+	"github.com/instill-ai/x/errmsg"
+)
+
+const (
+	apiKey  = "123"
+	errResp = `
+{
+  "error": {
+    "message": "Incorrect API key provided."
+  }
+}`
+)
+
+func TestComponent_Execute(t *testing.T) {
+	c := qt.New(t)
+	ctx := context.Background()
+
+	bc := base.Component{Logger: zap.NewNop()}
+	connector := Init(bc)
+
+	testcases := []struct {
+		name        string
+		task        string
+		path        string
+		contentType string
+	}{
+		{
+			name:        "text generation",
+			task:        textGenerationTask,
+			path:        messagesPath,
+			contentType: httpclient.MIMETypeJSON,
+		},
+	}
+
+	// TODO we'll likely want to have a test function per task and test at
```

Comment:
Question
Will it be possible to add the ok case?
We are probably able to mock client if you make it as interface in execution.
I think you can refer to the test cases in Slack component or Document component.

cc'd @donch1989 
I think it will be normal for the `execution struct` to have the `client` filed for all connecters like openai / slack / bigquery /....
However, I am not sure if it damages the current design.
So, I sync with you as well.